### DISCUSSION TITLE
adapt sample application EDS

### DISCUSSION THREAD

**Discussion by manggian** [EIPStackGroup/OpENer#517] [Q&A]
hello
given that I am actually successfully running the sample application with the sample EDS found in the repository, I would need to adapt the EDS file reducing the number of bytes on the  input and output assemblies from 32 bytes down to 8 bytes. Is the EDS file always parsed in the sample application so that the new data sizes are adapted, or shell I modify the sample application source  accordingly to the new EDS amount of data? thank you

**Comment by superPN** [comment]
You can modify the communication length described in the eds file by changing the number of "8,Param1," in Assem100 and Assem150 in the [Assembly] class in the eds file.
In the program, you only need to modify the length of the g_assembly_data064 and g_assembly_data096 arrays in the sampleapplication.c file to match the length in the eds file.
Then re-import the EPS file into Ethernet/IP Scanner and reconnect.
I have tried this method to extend the length of the communication from 32 bytes to 256 bytes.
